core/src/test/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorterSuite.java

```diff
@@ -234,7 +234,7 @@ public void testSortingEmptyArrays() throws Exception {
   public void spillingOccursInResponseToMemoryPressure() throws Exception {
     shuffleMemoryManager = new ShuffleMemoryManager(pageSizeBytes * 2);
     final UnsafeExternalSorter sorter = newSorter();
-    final int numRecords = 100000;
+    final int numRecords = (int) pageSizeBytes / 4;
```

Comment:
Fails for me consistently if I use `100000`.

    Failed tests: 
    org.apache.spark.util.collection.unsafe.sort.UnsafeExternalSorterSuite.spillingOccursInResponseToMemoryPressure(org.apache.spark.util.collection.unsafe.sort.UnsafeExternalSorterSuite)
      Run 1: UnsafeExternalSorterSuite.spillingOccursInResponseToMemoryPressure:243 
    Expected: a value equal to or greater than <1>
         got: <0>
    
      Run 2: UnsafeExternalSorterSuite.tearDown:150 expected:<0> but was:<69206016>